instrumentation/aws-lambda/aws-lambda-events-2.2/javaagent/build.gradle.kts

```diff
@@ -27,7 +27,7 @@ dependencies {
   // with lambda.
   // NB: 2.2.0 includes a class called SQSEvent but isn't usable due to it returning private classes
   // in public API.
-  library("com.amazonaws:aws-lambda-java-events:2.2.1")
+  library("com.amazonaws:aws-lambda-java-events:3.11.2")
```

Comment:
There `2.2` in the module and package name denotes the minimal supported library version. Usually we also compile and test against the against this version. When you need to support features from a newer version you can either update the minimal supported version (`3.1.0` or whatever is the first version that contains `APIGatewayV2HTTPEvent`) or create a new module for `3.1.0` and extract shared code into a common module. Usually we choose the second option as in javaagent we wish to support older versions, but as this instrumentation is mostly for library instrumentation users I think the first options is also ok. @open-telemetry/java-instrumentation-approvers any objections to bumping min supported version of `aws-lambda-java-events`?

Reply:
cc @open-telemetry/lambda-extension-approvers @rapphil any thoughts on removing OpenTelemetry Java agent support for older versions of `aws-lambda-java-events`?

here's our existing guidelines for dropping support for older versions: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/VERSIONING.md#dropping-support-for-older-library-versions

Reply:
I will investigate the new module workflow to get a sense of the effort involved.  The last modification to the `APIGatewayV2HTTP` APIs was in the 3.9.0 release and that is reflected in the updated code.

Reply:
> any thoughts on removing OpenTelemetry Java agent support for older versions of `aws-lambda-java-events`?

Do we really need to do that though? The javaagent instrumentation does not seem to reference the new types in any way; it's just the library instrumentation that uses them in the new wrapper class.

Reply:
That is a good point, I have rolled back the javaagent dependency to the previous version. The tests pass as expected.